docs/docs/api/Dispatcher.md

```diff
@@ -630,6 +631,29 @@ try {
 }
 ```
 
+#### Example 3 - Conditionally reading the body
+
+Remember to fully consume the body even in the case when it is not read.
+
+```js
+const { body, statusCode } = await client.request({
+  path: '/',
+  method: 'GET'
+})
+
+body.on('error', console.error) // prevent process from crashing on error
+
+if (statusCode === 200) {
+  const buffer = await body.arrayBuffer()
+  return buffer
+}
+
+for await (const _chunk of body) {
+  // force consumption of body to avoid memory leak
+}
+return null
+```
```

Comment:
```suggestion
```js
const { body, statusCode } = await client.request({
  path: '/',
  method: 'GET'
})

if (statusCode === 200) {
  return await body.arrayBuffer()
}

await body.dump()

return null
```
```

Reply:
@ronag @mcollina I think its important to keep the `body.on('error', console.error)` handler because any code added between the `client.request()` and the `await body` calls could throw and then it would be uncatchable.

Since this example is about conditional code, I think its best practice to keep it in there to avoid this footgun.

Reply:
Furthermore, I don't think `await body.dump()` is sufficient.

It says it defaults to 131072 but what if the body is larger than that?

https://github.com/nodejs/undici/blob/8e025d1cf5a453ff158cb1bd40ff6929e1b27bb1/lib/api/readable.js#L254

Reply:
Friendly ping @mcollina @ronag

Reply:
I created PR https://github.com/nodejs/undici/pull/3788

Reply:
> handler because any code added between the client.request() and the await body calls could throw and then it would be uncatchable.

That would be incorrect code anyway...

> It says it defaults to 131072 but what if the body is larger than that?

It will close the connection if the body is larger.